lib/src/main/java/com/knuddels/jtokkit/EncodingFactory.java

```diff
@@ -143,17 +154,15 @@ public static Map<byte[], Integer> loadMergeableRanks(String fileName) {
                 throw new IllegalStateException("Could not find " + fileName + " in resources");
             }
 
-            Map<byte[], Integer> mergeableRanks = new HashMap<>();
-            BufferedReader reader = new BufferedReader(new InputStreamReader(in, StandardCharsets.UTF_8));
+            Map<byte[], Integer> mergeableRanks = new LinkedHashMap<>(); // keep order to optimize collisions
+            BufferedReader reader = new BufferedReader(new InputStreamReader(in, UTF_8));
             String line;
             while ((line = reader.readLine()) != null) {
-                String[] parts = line.split("\\s+", 2);
-                if (parts.length != 2) {
-                    throw new IllegalStateException("Invalid line in " + fileName + ": " + line);
-                }
+                int firstSpaceIndex = line.indexOf(' ');
+                assert firstSpaceIndex != -1 : "Invalid line in " + fileName + ": " + line;
```

Comment:
Currently these assertions will never be verified or am I missing something?

As far as I know, we have to enable them explicitly via

```
junitPlatformTest {
    enableAssertions = true
}
```

which we should probably do :smile:

Reply:
They are enabled for tests:
```
assert false : "Invalid line in " + fileName + ": " + line;
```
results in test failure:
```
Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.AssertionError: Invalid line in /com/knuddels/jtokkit/cl100k_base.tiktoken: IQ== 0 [in thread "Test worker"]
	at com.knuddels.jtokkit.EncodingFactory.loadMergeableRanks(EncodingFactory.java:166)
	```

Reply:
Do you want me to change it to throwing an exception?

Reply:
No, I'm perfectly fine with it being an assert keyword. Since we pre-package the vocabulary at build time, the failing test is enough in my opinion